doc/pull-requests.md

```diff
@@ -0,0 +1,155 @@
+  - A request on the design review should be an explicit comment.
+  - Such PR should be marked as a draft or with the WIP prefix.
+
+## Requesting a Review
+
+<a name="review-reqs"></a>
+- [1.](#review-reqs) A review can be requested when:
+  - [The PR template](#pr-template) is filled in.
+  - Changes are thoroughly tested by an author.
+  - Changes thoroughly reviewed following the [review checklist](#review-checklist) by an author.
+<a name="review-request-gh"></a>
+- [2.](#review-request-gh) A review can be requested explicitly using GitHub: https://help.github.com/en/articles/requesting-a-pull-request-reviewed
+<a name="review-request-comment"></a>
+- [3.](#review-request-comment) A review can be also requested as a comment from any GitHub users.
+  - For example to invite the person who originally filed an issue for testing.
+
+## Review Checklist
+
+<a name="checklist-build-and-test"></a>
+- [1.](#checklist-build-and-test) The new code is built and tested according to the `How to test` section of a PR description.
+<a name="checklist-project-org"></a>
+- [2.](#checklist-project-org) The new code is aligned with the [project organization](https://github.com/theia-ide/theia/wiki/Code-Organization) and [coding conventions](https://github.com/theia-ide/theia/wiki/Coding-Guidelines).
+<a name="checklist-changelog"></a>
+- [3.](#checklist-changelog) [Changelog](https://github.com/theia-ide/theia/blob/master/CHANGELOG.md) is updated.
+<a name="checklist-breaking-changes"></a>
+- [4.](#checklist-breaking-changes) Breaking changes are justified and recorded in the [changelog](https://github.com/theia-ide/theia/blob/master/CHANGELOG.md).
+<a name="checklist-dependencies"></a>
+- [5.](#checklist-dependencies) New dependencies are justified and [verified](https://github.com/theia-ide/theia/wiki/Registering-CQs#wip---new-ecd-theia-intellectual-property-clearance-approach-experimental).
+<a name="checklist-copied-code"></a>
+- [6.](#checklist-copied-code) Copied code is justified and [approved via a CQ](https://github.com/theia-ide/theia/wiki/Registering-CQs#case-3rd-party-project-code-copiedforked-from-another-project-into-eclipse-theia-maintained-by-us).
+<a name="checklist-copyright"></a>
+- [7.](#checklist-copyright) Each new file has proper copyright with the current year and the name of contributing entity (individual or company).
+<a name="checklist-sign-off"></a>
+- [8.](#checklist-sign-off) Commits are signed-off: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#sign-your-work.
+<a name="checklist-commit-history"></a>
+- [9.](#checklist-commit-history) Commit history is rebased on master and contains only meaningful commits and changes (less are usually better).
+  - For example, use `git pull -r` or `git fetch && git rebase` to pick up changes from the master.
+
+## Reviewing
+
+<a name="reviewing-template"></a>
+- [1.](#eviewing-template) Reviewers should check that a PR has a [proper description](#pr-template).
```

Comment:
Suggestion: also make it clear that the commit(s) should have a proper description (at minimum equivalent to `What it does` from the PR template. Other info from filled-in template could be nice-to-have as well). This is missed sometimes and makes it harder to understand what a commit does without digging-in the code.

Reply:
Maybe this would fit better in the "review checklist" section, around \#8 / \#9

Reply:
@marcdumais-work i think it should go to review checklist, could you suggest wording.

Also @benoitf suggested to use conventional commits. I think it is a good idea in the sense that we should automate as much as possible from this checklist. With conventional commits we will need only ask to provide proper commit messages and then we can generate changelogs from them. I know that electon follow such approach. It could be worth to look into after landing this PR.

Reply:
>  could you suggest wording.

- [10.]() Each commit has meaningful title and a body that explains what it does. One can take inspiration from the "what it does" section from the PR.

> Also @benoitf suggested to use conventional commits. [...] It could be worth to look into after landing this PR.

:+1: